etcdserver/auth/auth.go

```diff
@@ -192,6 +182,10 @@ func (s *Store) CreateOrUpdateUser(user User) (out User, created bool, err error
 
 func (s *Store) CreateUser(user User) (User, error) {
 	u, err := s.createUserInternal(user)
+	// Attach root role to root user.
+	if u.User == "root" {
```

Comment:
current logic is that stored root user always doesn't have root role, and it attaches root role when read.

Another way is that root user is always stored with root role.

I am fine with both, but I don't want to do a half-mix one.